contrib/pg_tde/src/catalog/tde_principal_key.c

```diff
@@ -370,12 +370,59 @@ set_principal_key_with_keyring(const char *key_name, const char *provider_name,
 void
 xl_tde_perform_rotate_key(XLogPrincipalKeyRotate *xlrec)
 {
+	TDEPrincipalKey *curr_principal_key;
+	TDEPrincipalKey *new_principal_key;
+	GenericKeyring *new_keyring;
+	const KeyInfo *keyInfo = NULL;
```

Comment:
This one is a bit of an outlier. I don't mind the `const` shouldn't the rest be too? Or otherwise none of them should be. Also the default value seems to serve no purpose.